pkg/egresspolicy/egresspolicy.go

```diff
@@ -0,0 +1,566 @@
+	var endpointSelectorList []api.EndpointSelector
+	var dstCidrList []*net.IPNet
+
+	allowAllNamespacesRequirement := slim_metav1.LabelSelectorRequirement{
+		Key:      k8sConst.PodNamespaceLabel,
+		Operator: slim_metav1.LabelSelectorOpExists,
+	}
+	name := cenp.ObjectMeta.Name
+
+	if name == "" {
+		return nil, fmt.Errorf("CiliumEgressNATPolicy must have a name")
+	}
+
+	for _, cidrString := range cenp.Spec.DestinationCIDRs {
+		_, cidr, err := net.ParseCIDR(cidrString)
+		if err != nil {
+			log.WithError(err).Warnf("CiliumEgressNATPolicy %v error parsing cidr", name)
+			continue
+		}
+		dstCidrList = append(dstCidrList, cidr)
+	}
+
+	for _, egressRule := range cenp.Spec.Egress {
+		if egressRule.NamespaceSelector != nil {
+			prefixedNsSelector := egressRule.NamespaceSelector
+			matchLabels := map[string]string{}
+			// We use our own special label prefix for namespace metadata,
+			// thus we need to prefix that prefix to all NamespaceSelector.MatchLabels
+			for k, v := range egressRule.NamespaceSelector.MatchLabels {
+				matchLabels[policy.JoinPath(k8sConst.PodNamespaceMetaLabels, k)] = v
+			}
+			egressRule.NamespaceSelector.MatchLabels = matchLabels
+
+			// We use our own special label prefix for namespace metadata,
+			// thus we need to prefix that prefix to all NamespaceSelector.MatchLabels
+			for i, lsr := range egressRule.NamespaceSelector.MatchExpressions {
+				lsr.Key = policy.JoinPath(k8sConst.PodNamespaceMetaLabels, lsr.Key)
+				egressRule.NamespaceSelector.MatchExpressions[i] = lsr
+			}
+
+			// Empty namespace selector selects all namespaces (i.e., a namespace
+			// label exists).
+			if len(egressRule.NamespaceSelector.MatchLabels) == 0 && len(egressRule.NamespaceSelector.MatchExpressions) == 0 {
+				egressRule.NamespaceSelector.MatchExpressions = []slim_metav1.LabelSelectorRequirement{allowAllNamespacesRequirement}
+			}
+
+			endpointSelectorList = append(
+				endpointSelectorList,
+				api.NewESFromK8sLabelSelector("", prefixedNsSelector, egressRule.PodSelector))
+		} else if egressRule.PodSelector != nil {
+			endpointSelectorList = append(
+				endpointSelectorList,
+				api.NewESFromK8sLabelSelector("", egressRule.PodSelector))
+		} else {
+			return nil, fmt.Errorf(
+				"CiliumEgressNATPolicy %v cannot have both nil namespace selector and nil pod selector",
+				name)
+		}
+	}
+
+	return &EgressPolicyConfig{
+		uid:               cenp.UID,
+		endpointSelectors: endpointSelectorList,
+		dstCIDRs:          dstCidrList,
+		egressIP:          net.ParseIP(cenp.Spec.EgressSourceIP).To4(),
+		id: types.NamespacedName{
+			Name: name,
+		},
+	}, nil
+}
+
+func NewEgressPolicyManager() *Manager {
+	return &Manager{
+		policyPods:    make(map[podID][]policyID),
+		policyConfigs: make(map[policyID]*EgressPolicyConfig),
+		podDataStore:  make(map[podID]*podMetadata),
+	}
+}
+
+func (manager *Manager) RegisterGetStores(sg StoreGetter) {
+	manager.storeGetter = sg
+}
+
+// Event handlers
```

Comment:
As mentioned in a previous comment that I can't find anymore, please extract the manager out to a separate file, similar to CLRP